### ISSUE TITLE
Unable to Add Orbstack Machine to Connections

### ISSUE THREAD

**Issue by oliv10** [wavetermdev/waveterm#127]
**Describe the bug**
I use another piece of software called [Orbstack](https://orbstack.dev/). One really cool thing it allows you to do is create what is basically a VM that runs on your own machine. And by default, they are accessible by SSH to your machine only. However, after reading the release notes for 0.5.1, I assume the issue is that the REGEX for the connection only allows for one '@' symbol. Just like how it wouldn't allow for dots or underscores which _was_ an issue for me as well.

**To Reproduce**
1. Install Orbstack
2. Configure Linux Machine
3. Configure connection to Orbstack machine on Waveterm

**Expected behavior**
Creates an SSH connects to the specified host.
<img width="684" alt="Screenshot 2023-12-06 at 12 30 02 PM" src="https://github.com/wavetermdev/waveterm/assets/18117030/d4cfcaf5-4c4f-42b8-b306-29dde4534081">

**Screenshots**
<img width="475" alt="image" src="https://github.com/wavetermdev/waveterm/assets/18117030/be062343-d571-4fa6-a1e9-b58ea237b08a">

**Desktop (please complete the following information):**
 - OS: MacOS
 - Version 0.5.1

**Additional context**
Need to have [Orbstack](https://orbstack.dev/) installed and configured.

**Comment by oliv10** [comment]
```
# AUTO-GENERATED BY ORBSTACK. DO NOT EDIT.
# To make changes, add or override hosts at the top of ~/.ssh/config

Host orb
  Hostname 127.0.0.1
  Port 32222
  # SSH user syntax:
  #   <container>@orb to connect to <container> as the default user (matching your macOS user)
  #   <user>@<container>@orb to connect to <container> as <user>
  # Examples:
  #   ubuntu@orb: container "ubuntu", user matching your macOS user
  #   root@fedora@orb: container "fedora", user "root"
  User default
  # replace or symlink ~/.orbstack/ssh/id_ed25519 file to change the key
  IdentityFile ~/.orbstack/ssh/id_ed25519
  # only use this key
  IdentitiesOnly yes
  ProxyCommand '/Applications/OrbStack.app/Contents/Frameworks/OrbStack Helper.app/Contents/MacOS/OrbStack Helper' ssh-proxy-fdpass /Users/USERNAME
  ProxyUseFdpass yes
```

For more context, Orbstack auto generates its own config file and its then incuded this in the ~/.ssh/config so it can function.

**Comment by sawka** [comment]
Thanks for submitting!  Hadn't heard of Orb before, but it looks cool.  I don't see any big reason why we couldn't loosen up the syntax to allow for a username with an "@" in it.  If it works with ssh (which your screenshot clearly showed it does), it should be possible to support.  Will try to get this in the v0.5.2 release.

Will mark as an enhancement (as opposed to a bug) since I feel like usernames with at-signs aren't standard :)

**Comment by oliv10** [comment]
Awesome, glad to see that! One of my co-workers found this about a week ago and it works really nicely with our labs workflow so I will probably have some more ideas or bugs to throw at you in time.

**Comment by oneirocosm** [comment]
close as this should be fixed by #129
